chore(postinst): Delete unnecessary postinst code

holmanb opened the pull request:
<!--
Thank you for submitting a PR to cloud-init!

To ease the process of reviewing your PR, do make sure to complete the following checklist **before** submitting a pull request.

- [ ] I have signed the CLA: https://ubuntu.com/legal/contributors
- [ ] I have added my Github username to ``tools/.github-cla-signers``
- [ ] I have included a comprehensive commit message using the guide below
- [ ] I have added unit tests to cover the new behavior under ``tests/unittests/``
  - Test files should map to source files i.e. a source file ``cloudinit/example.py`` should be tested by ``tests/unittests/test_example.py``
  - Run unit tests with ``tox -e py3``
- [ ] I have kept the change small, avoiding unnecessary whitespace or non-functional changes.
- [ ] I have added a reference to issues that this PR relates to in the PR message (Refs GH-1234, Fixes GH-1234)
- [ ] I have updated the documentation with the changed behavior.
  - If the change doesn't change the user interface and is trivial, this step may be skipped.
  - Cloud-config documentation is generated from the jsonschema.
  - Generate docs with ``tox -e docs``.
-->


## Proposed Commit Message
```
chore(postinst): Delete unnecessary postinst code
```

## Additional Context

1) Delete version gated code
- most recent version gate was 22.1-14-g2e17a0d6-0ubuntu1~22.04.5
- If someone somehow upgraded a Jammy instance to noble without getting the newer Jammy cloud-init package available in jammy-updates, the fix in the postinst would still run since this code exists in all releases of noble.

2) Delete old preseed code. Debian doesn't use our postinst, and we don't use preseed, so just drop it -> we can probably do this with the maas preseed code too but that requires more investigation


## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

TheRealFalcon commented on debian/cloud-init.postinst:
```diff
@@ -109,22 +109,6 @@ for k in sys.argv[2:]:
    db_unregister "cloud-init/maas-metadata-credentials" || :
 }
 
-handle_preseed_local_cloud_config() {
```
The call site needs to be removed too

TheRealFalcon commented on debian/cloud-init.postinst:
```diff
@@ -390,23 +216,6 @@ EOF
-   fix_lp1889555 "$2"
-   change_cloud_init_output_log_permissions "$2"
-
-   rename_hook_hotplug_udev_rule
```
Did you mean to include this one? The definition hasn't been deleted and this is a relatively recent addition.

holmanb replied:
No I didn't - fixed.

TheRealFalcon replied:
I'm not seeing it.

TheRealFalcon reviewed (changes requested):
I left some inline comments.

If we're confident there's no way we can install with preseed anymore, than I'm not sure why we can't also delete the MAAS preseed code.

holmanb commented:
> If we're confident there's no way we can install with preseed anymore, than I'm not sure why we can't also delete the MAAS preseed code.

+1 agreed

@TheRealFalcon I think that I've addressed all of your comments.

TheRealFalcon reviewed (changes requested):
Two more inline.

TheRealFalcon commented:
@holmanb , just because it is showing up under the resolved comments, I wanted to make sure you can see my latest comment: https://github.com/canonical/cloud-init/pull/5487#discussion_r1686522844
